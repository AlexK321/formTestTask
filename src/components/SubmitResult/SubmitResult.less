.result-block{
  background-color: white;
  border-radius: 4px;
  padding: 24px 40px!important;
  width: 100%;
  min-height: 300px;
  display: flex;
  justify-content: center;
  align-items: center;
}  