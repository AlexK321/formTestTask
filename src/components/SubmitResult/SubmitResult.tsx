import React, { FC } from 'react';
import './SubmitResult.css';

interface ParamTypes {
  hasError: boolean;
}

const SubmitResult: FC<ParamTypes> = ({ hasError }) => {
  const description = hasError ? `Данные формы не отправлены` : `Данные формы отправлены успешно`;

  return (
    <div className="result-block">
      <h2>{description}</h2>
    </div>
  );
};

export default SubmitResult;
