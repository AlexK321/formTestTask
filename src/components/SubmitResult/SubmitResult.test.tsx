import React from 'react';
import { shallow } from 'enzyme';
import SubmitResult from './SubmitResult';
/*eslint-disable */
describe('test SubmitResult component', () => {
  const setUp = (props?: any) => shallow(<SubmitResult {...props} />)
  let component: any = null;
  beforeEach(() => {
    component = setUp();
  })

  it('check existence component', () => {
    expect(component).toBeTruthy();
  });

  it('should contain .result-block', () => {
    const wrapper = component.find('.result-block')
    expect(wrapper.length).toBe(1);
  });

  it('snapshot component', () => {
    expect(component).toMatchSnapshot();
  });
})

describe('test props', () => {
  it('should render negative response', () => {
    const response = true
    const responseMessage = 'Данные формы не отправлены'
    const component = shallow(<SubmitResult hasError={response} />)
    const description = component.find('h2');
    expect(description.text()).toBe(responseMessage);
  });

  it('should render positive response', () => {
    const response = false
    const responseMessage = 'Данные формы отправлены успешно'
    const component = shallow(<SubmitResult hasError={response} />)
    const description = component.find('h2');
    expect(description.text()).toBe(responseMessage);
  });
});
