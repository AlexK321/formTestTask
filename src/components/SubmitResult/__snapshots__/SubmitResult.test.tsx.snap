// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SubmitResult component snapshot component 1`] = `
<div
  className="result-block"
>
  <h2>
    Данные формы отправлены успешно
  </h2>
</div>
`;
